plugin-guide.md

```diff
@@ -42,11 +54,31 @@ The plugins can be associated with targets in the server configuration.
 Consumable resources _may_ request that submitters provide information or
 operate on the existence of a workflow run.
 
+Consumable resources are long-running. Whenever Vidarr attempts to run a
+workflow, it will consult the consumable resources to see if there is capacity
+to run the workflow (the `request` method). At that point the consumable
+resource must make a decision as to whether the workflow can proceed. Once the
+workflow has finished running (successfully or not), Vidarr will `release` the
+resource so that it can be used again. When Vidarr restarts, any running
+workflows will be called with `recover` to indicate that the resource is being
+used and the resource cannot stop the workflow even if the resource is
+over-capacity.
+
+Consumable resources can request data from the user, if desired. The
+`inputFromSubmitter` can return an empty optional to indicate that no
+information is required or can indicate the name and type of information that
+is required. The `request` method will contain a copy of this information,
+encoded as JSON, if the submitter provided it. The JSON data has been
+type-checked by Vidarr, so it should be safe to convert to the expected type
+using Jackson.
+
 # Input Provisioners
 Input provisioners implement `ca.on.oicr.gsi.vidarr.InputProvisionerProvider`
 and `ca.on.oicr.gsi.vidarr.InputProvisioner`. These plugins are responsible for
 taking files from existing workflows or provided by the user and generating a
-file path that a workflow can use.
+file path that a workflow can use. Input provisioners can choose to handle only
+some kinds of input data (files vs directories) and the system administrator
+can choose multiple provisoners to handle both.
 
 This plugin and the workflow plugins must have a mutual understanding of what a
 file path means. That is somewhat the responsibility of the system
```

Comment:
Line 86: is 'provision in plugin' a synonym for InputProvisioner? I would find this easier to follow if this language was consistent